src/arkode/arkode_arkstep.c

```diff
@@ -3654,6 +3858,63 @@ int arkStep_GetOrder(ARKodeMem ark_mem)
   return step_mem->q;
 }
 
+/* -----------------------------------------------------------------------------
+ * arkStep_TryStep
+ *
+ * Attempts one internal time step using ARKStepEvolve
+ * ---------------------------------------------------------------------------*/
+
+int arkStep_TryStep(void* arkode_mem, sunrealtype tstart, sunrealtype tstop,
```

Comment:
The body of this function was taken from `ARKBraid_TakeStep` and some of the behavior is specific to XBraid's use (having to reset before every step and use a specific step size) and I'm not sure if it's how we want to `SUNStepper` to behave. This is related to my question above about having a stepper advance from its last reached state.

1. If `TryStep` is called in a loop (i.e., `SUNStepper` could provide an implementation of `Advance` that calls `TryStep` as needed) the call to `Reset` could lead to unnecessary RHS evaluations.
2. Do we want the step size to be determined by `tstart` and `tstop` or internally by the stepper?